services/src/main/java/io/grpc/services/ChannelzProtoUtil.java

```diff
@@ -168,29 +166,31 @@ static Address toAddress(SocketAddress address) {
   }
 
   static SocketData extractSocketData(SocketStats socketStats) {
-    TransportStats s = socketStats.data;
-    return SocketData
-        .newBuilder()
-        .setStreamsStarted(s.streamsStarted)
-        .setStreamsSucceeded(s.streamsSucceeded)
-        .setStreamsFailed(s.streamsFailed)
-        .setMessagesSent(s.messagesSent)
-        .setMessagesReceived(s.messagesReceived)
-        .setKeepAlivesSent(s.keepAlivesSent)
-        .setLastLocalStreamCreatedTimestamp(
-            Timestamps.fromNanos(s.lastLocalStreamCreatedTimeNanos))
-        .setLastRemoteStreamCreatedTimestamp(
-            Timestamps.fromNanos(s.lastRemoteStreamCreatedTimeNanos))
-        .setLastMessageSentTimestamp(
-            Timestamps.fromNanos(s.lastMessageSentTimeNanos))
-        .setLastMessageReceivedTimestamp(
-            Timestamps.fromNanos(s.lastMessageReceivedTimeNanos))
-        .setLocalFlowControlWindow(
-            Int64Value.newBuilder().setValue(s.localFlowControlWindow).build())
-        .setRemoteFlowControlWindow(
-            Int64Value.newBuilder().setValue(s.remoteFlowControlWindow).build())
-        .addAllOption(toSocketOptionsList(socketStats.socketOptions))
-        .build();
+    SocketData.Builder builder = SocketData.newBuilder();
+    if (socketStats.data != null) {
+      TransportStats s = socketStats.data;
+      builder
+          .setStreamsStarted(s.streamsStarted)
+          .setStreamsSucceeded(s.streamsSucceeded)
+          .setStreamsFailed(s.streamsFailed)
+          .setMessagesSent(s.messagesSent)
+          .setMessagesReceived(s.messagesReceived)
+          .setKeepAlivesSent(s.keepAlivesSent)
+          .setLastLocalStreamCreatedTimestamp(
+              Timestamps.fromNanos(s.lastLocalStreamCreatedTimeNanos))
+          .setLastRemoteStreamCreatedTimestamp(
+              Timestamps.fromNanos(s.lastRemoteStreamCreatedTimeNanos))
+          .setLastMessageSentTimestamp(
+              Timestamps.fromNanos(s.lastMessageSentTimeNanos))
+          .setLastMessageReceivedTimestamp(
+              Timestamps.fromNanos(s.lastMessageReceivedTimeNanos))
+          .setLocalFlowControlWindow(
+              Int64Value.newBuilder().setValue(s.localFlowControlWindow).build())
```

Comment:
The interior builds() are unnecessary I think

Reply:
Changed to `.of()`